SUS-4591: Allow to send MediaWiki logs to stdout as JSON

mszabo-wikia opened the pull request:
https://wikia-inc.atlassian.net/browse/SUS-4591

mszabo-wikia committed:
SUS-4591: Allow to send MediaWiki logs to stdout as jSON

pwojcik86 reviewed (approved):
Is this just for db logs?

macbre commented on lib/Wikia/tests/Logger/StatusProcessorTest.php:
```diff
@@ -0,0 +1,43 @@
+		$formatted = call_user_func( $this->statusProcessor, $record );
+
+		$this->assertTrue( $formatted['context']['a_status']['is_ok'] );
+		$this->assertEquals( 1791, $formatted['context']['a_status']['value'] );
```
:+1:

pwojcik86 committed:
Merge branch 'dev' into SUS-4591

mszabo-wikia committed:
SUS-4591: Fix k8s logging

mszabo-wikia committed:
SUS-4591: Pipe logs

pwojcik86 committed:
Merge branch 'dev' into SUS-4591

pwojcik86 committed:
SUS-4591 sends logs to socket

pwojcik86 committed:
SUS-4591 use external log output

pwojcik86 committed:
SUS-4591 use socat to handle parallel connections

macbre committed:
Merge branch 'dev' into SUS-4591

macbre committed:
WikiaLogger: make it compatible with S3AuditLogger class

macbre committed:
LoggerFactoryTest: fix for changes in LoggerFactory

macbre committed:
docker: change the context where dev/Dockerfile is built

macbre committed:
LocalSettings | throw RuntimeException, ConfigException is not yet available

macbre committed:
Docker: update README.md on how to run eval.php in container

macbre committed:
nginx: handle redirect from domain-only URLs

macbre committed:
dev/docker-compose: add logger image and use it to log JSON messages

logger_1     | {"appname":"mediawiki","@timestamp":"2018-06-13T07:56:17.836674+00:00","@message":"LoadBalancer::reallyOpenConnection","@fields":{"app_name":"mediawiki","app_version":"","datacenter":"poz","environment":"dev","wiki_dbname":"muppet","wiki_id":"831","http_url_path":"/wiki/Muppet_Wiki","http_method":"GET","http_url_domain":"muppet.wikia-local.com","http_url":"http://muppet.wikia-local.com/wiki/Muppet_Wiki","client_ip":"192.168.16.1","span_id":"243b2456-66b3-4ba7-897a-f3df1a97863b","trace_id":"0f83d04e-0238-4d2a-88a4-1afd550d7efc"},"@context":{"caller":"ExternalStoreDB:getSlave","host":"geo-db-dev-db-slave.query.consul","db_name":"blobs20121","db_user":"mw_only"}}

macbre commented:
MediaWiki's JSON-formatted logs are now "available" via logger container:

```
logger_1     | {"appname":"mediawiki","@timestamp":"2018-06-13T07:56:17.836674+00:00","@message":"LoadBalancer::reallyOpenConnection","@fields":{"app_name":"mediawiki","app_version":"","datacenter":"poz","environment":"dev","wiki_dbname":"muppet","wiki_id":"831","http_url_path":"/wiki/Muppet_Wiki","http_method":"GET","http_url_domain":"muppet.wikia-local.com","http_url":"http://muppet.wikia-local.com/wiki/Muppet_Wiki","client_ip":"192.168.16.1","span_id":"243b2456-66b3-4ba7-897a-f3df1a97863b","trace_id":"0f83d04e-0238-4d2a-88a4-1afd550d7efc"},"@context":{"caller":"ExternalStoreDB:getSlave","host":"geo-db-dev-db-slave.query.consul","db_name":"blobs20121","db_user":"mw_only"}}
```

macbre committed:
dev/k8s: update README
